src/Catalyst.Core.Modules.Rpc.Client/IO/Observers/PeerListResponseObserver.cs

```diff
@@ -0,0 +1,49 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using Catalyst.Abstractions.IO.Messaging.Correlation;
+using Catalyst.Abstractions.IO.Observers;
+using Catalyst.Core.Lib.Rpc.IO;
+using Catalyst.Protocol.Peer;
+using Catalyst.Protocol.Rpc.Node;
+using DotNetty.Transport.Channels;
+using Serilog;
+
+namespace Catalyst.Core.Modules.Rpc.Client.IO.Observers
+{
+    /// <summary>
+    ///     Handles the Peer list response from the node
+    /// </summary>
+    /// <seealso cref="IRpcResponseObserver" />
+    public sealed class PeerListResponseObserver
+        : RpcResponseObserver<GetPeerListResponse>
+    {
+        public PeerListResponseObserver(ILogger logger)
+            : base(logger) { }
+
+        protected override void HandleResponse(GetPeerListResponse getPeerListResponse,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)